File: 09_2차원_배열/Problem_03.cpp
/* 09단계 [ 2차원 배열 ] */
// 제목 : 행렬 덧셈
// 문제 번호 : 2738

#include <iostream>
using namespace std;
int main()
{
    bool arr[100][100] = {false};
    int N, x, y, cnt = 0;
    cin >> N;
    for (int i = 0; i < N; i++)
    {
        cin >> x >> y;
        for (int j = x; j < 10 + x; j++)
            for (int k = y; k < 10 + y; k++)
                arr[j][k] = true;
    }
    for (int i = 0; i < 100; i++)
        for (int j = 0; j < 100; j++)
            if (arr[i][j])
                cnt++;
    cout << cnt << "\n";
}

File: Python/09_2차원_배열/Problem_03.py
N = int(input())
cnt = 0
arr = [[False for _ in range(100)] for _ in range(100)]
for _ in range(N):
    x, y = map(int, input().split())
    for i in range(10):
        for j in range(10):
            arr[i + x][j + y] = True
for i in range(100):
    for j in range(100):
        if arr[i][j]:
            cnt += 1
print(cnt)
